django/contrib/admin/options.py

```diff
@@ -884,7 +884,12 @@ def construct_search(field_name):
             if field_name.startswith('^'):
                 return "%s__istartswith" % field_name[1:]
             elif field_name.startswith('='):
-                return "%s__iexact" % field_name[1:]
+                field_name = field_name[1:]
+                # Only strings care about case-sensitivity.
+                if isinstance(queryset.model._meta.get_field(field_name),
```

Comment:
@charettes 
`get_field` is failing to make a reverse lookup for the fields `recommender__titletranslation__text` and `titletranslation__text` on the `Recommendation` model for [this model structure](https://github.com/django/django/blob/master/tests/admin_views/models.py#L385) and is raising the error `Recommendation has no field named 'titletranslation__text'`. How can I resolve it ?